build/lib/src/builder/builder.dart

```diff
@@ -1,21 +1,24 @@
-// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 import 'dart:async';
 
-import '../asset/id.dart';
 import 'build_step.dart';
 
 /// The basic builder class, used to build new files from existing ones.
 abstract class Builder {
   /// Generates the outputs for a given [BuildStep].
   Future build(BuildStep buildStep);
 
-  /// Declares all potential output assets given [inputId]. If an empty [List]
-  /// is returned then the asset will never be provided to [build].
+  /// Configuration for which files will be created based on the file extension
+  /// of inputs.
   ///
-  /// Note: Reading the contents of [inputId] is not supported during this
-  /// phase. You can however choose later on not to actually output assets which
-  /// you return here, but no other [Builder] will be able to output them.
-  List<AssetId> declareOutputs(AssetId inputId);
+  /// All input sources matching any key in this map will be passed as build
+  /// step to this builder. Only files with the same basename and an extension
+  /// from the values in this map are expected as outputs. If an empty key
+  /// exists, all inputs are considered matching. A builder must always return
+  /// the same configuration. Typically this will be `const` but may vary based
+  /// on build arguments. Most builders will use a single input extension and
+  /// one or more output extensions.
+  Map<String, List<String>> get buildExtensions;
```

Comment:
Yeah I don't know. `outputExtensions` feels less accurate to me, `extensionMapping` just repeats the type in the name... Will leave it as is for now and see if we come up with anything else before publishing